actions.js

```diff
@@ -255,4 +255,95 @@ btnProject7.addEventListener('click', () => {
+const email = document.getElementById('email');
+const message = document.getElementById('message');
+const formvalidmsg = document.getElementById('formvalidmsg');
+
+function isValidEmail(email) {
+  const regexValidate = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return regexValidate.test(String(email).toLowerCase());
+}
+
+// eslint-disable-next-line no-unused-vars
+fullname.addEventListener('input', (event) => {
+  if (fullname.validity.valid) {
+    formvalidmsg.innerHTML = '';
+    fullname.style.backgroundColor = 'rgb(215, 255, 225)';
+    fullname.style.border.color = 'rgb(215, 255, 225)';
+  } else {
+    formvalidmsg.innerHTML = 'Name should be 2 to 30 characters long!';
+    formvalidmsg.style.color = 'red';
+    fullname.style.backgroundColor = 'rgb(255, 204, 204)';
+    fullname.style.border.color = 'rgb(255, 204, 204)';
+  }
+});
+
+// eslint-disable-next-line no-unused-vars
+email.addEventListener('input', (event) => {
+  if (isValidEmail(email.value)) {
+    if (email.value !== email.value.toLowerCase()) {
+      formvalidmsg.innerHTML = 'Email should be all lowercase!';
+      formvalidmsg.style.color = 'red';
+      email.style.backgroundColor = 'rgb(255, 204, 204)';
+      email.style.border.color = 'rgb(255, 204, 204)';
+    } else {
+      formvalidmsg.innerHTML = '';
+      email.style.backgroundColor = 'rgb(215, 255, 225)';
+      email.style.border.color = 'rgb(215, 255, 225)';
+    }
+  } else {
+    formvalidmsg.innerHTML = 'Email is not in a valid format!';
+    formvalidmsg.style.color = 'red';
+    email.style.backgroundColor = 'rgb(255, 204, 204)';
+    email.style.border.color = 'rgb(255, 204, 204)';
+  }
+});
+
+// eslint-disable-next-line no-unused-vars
+message.addEventListener('input', (event) => {
+  if (message.validity.valid) {
+    formvalidmsg.innerHTML = '';
+    message.style.backgroundColor = 'rgb(215, 255, 225)';
+    message.style.border.color = 'rgb(215, 255, 225)';
+  } else {
+    formvalidmsg.innerHTML = 'Message should be 1 to 500 characters long!';
+    formvalidmsg.style.color = 'red';
+    message.style.backgroundColor = 'rgb(255, 204, 204)';
+    message.style.border.color = 'rgb(255, 204, 204)';
+  }
+});
+form.addEventListener('submit', (event) => {
+  if (!message.validity.valid) {
+    formvalidmsg.innerHTML = 'Message should be 1 to 500 characters long!';
+    formvalidmsg.style.color = 'red';
+    message.style.backgroundColor = 'rgb(255, 204, 204)';
+    message.style.border.color = 'rgb(255, 204, 204)';
+    event.preventDefault();
+  }
+  if (!isValidEmail(email.value)) {
+    formvalidmsg.innerHTML = 'Email is not in a valid format!';
+    formvalidmsg.style.color = 'red';
+    email.style.backgroundColor = 'rgb(255, 204, 204)';
+    email.style.border = 'rgb(255, 204, 204)';
+    event.preventDefault();
+  } else if (email.value !== email.value.toLowerCase()) {
```

Comment:
Hey @acolombo1 
Good job so far However, I tested your contact form, it's accepting email with uppercase as well which it shouldn't. Kindly make sure that the email field will only accept text with all lowercase letters.

Reply:
@zainsadaqat : I've been testing it a lot and there's no way it passes with an uppercase in email. You can check it also live at: https://acolombo1.github.io/Portfolio/. I really don't know what issue it could be. If you can provide me an example email input, or maybe in your operating system it works differently...
If you tell me the problem persists I'll see to re-write the code.

Reply:
Take a look at https://www.loom.com/share/ce7d3d13364746c3be8a979242096819

I would really like to know what the issue is.